pilot/pkg/xds/workloadentry.go

```diff
@@ -121,6 +130,49 @@ func (sg *InternalGen) QueueUnregisterWorkload(proxy *model.Proxy) {
 	}, features.WorkloadEntryCleanupGracePeriod)
 }
 
+// UpdateWorkloadEntryHealth updates the associated WorkloadEntries health status
+// based on the corresponding health check performed by istio-agent.
+func (sg *InternalGen) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthEvent) {
+	if !features.WorkloadEntryHealthChecks {
+		return
+	}
+	// we assume that the workload entry exists
+	// if auto registration does not exist, try looking
+	// up in NodeMetadata
+	entryName := autoregisteredWorkloadEntryName(proxy)
+	if entryName == "" {
+		// todo make this env var a constant somewhere
+		name, ok := proxy.Metadata.ProxyConfig.ProxyMetadata["WORKLOAD_ENTRY_NAME"]
+		if !ok {
+			// at this point we could not find the WorkloadEntry Name,
+			// just return.
+			adsLog.Errorf("unable to derive WorkloadEntry Name, blank name")
```

Comment:
```suggestion
			adsLog.Errorf("unable to derive WorkloadEntry for health update for %v", proxy.ID)
```

Reply:
We should make sure all logs have enough context about what it was doing and what connection it was